### PR TITLE
CLOUDP-268554 Update atlas diagnostics command to add more information

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Update atlas diagnostics command to add more information

_Jira ticket:_ CLOUDP-268554

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/container/docker.go [hunk 1]
```diff
@@ -392,13 +393,32 @@ func (e *dockerImpl) ImageList(ctx context.Context, references ...string) ([]Ima
...
-	if err := json.Unmarshal(buf, &result); err != nil {
+	result, err := readJsonl[Image](bytes.NewBuffer(buf))
+	if err != nil {
 		return nil, err
 	}
+
 	return result, nil
 }
 
+func readJsonl[T any](r io.Reader) ([]T, error) {
```
Comment (blva) [inline internal/container/docker.go:404]: is it possible to add a unit test for this?

File: internal/cli/deployments/diagnostics.go [hunk 2]
```diff
@@ -59,34 +64,51 @@ func (opts *diagnosticsOpts) Run(ctx context.Context) error {
 		Engine: opts.ContainerEngine.Name(),
 	}
 
-	images, err := opts.ContainerEngine.ImageList(ctx)
+	cores, err := cpu.Counts(true)
```
Comment (blva) [inline internal/cli/deployments/diagnostics.go:67]: do we call diagnostics on e2e test failures? would be a nice approach to ensure we call it in some tests.. overall I think as this command has grown, we should add unit/e2e tests for it, but can be in another ticket!

**Review by blva — DISMISSED** [review]
LGTM, comments not blocking

File: internal/cli/deployments/diagnostics.go (hunk 2 above)
Comment (fmenezes) [reply internal/cli/deployments/diagnostics.go:67]: Yes we do

Comment (fmenezes) [reply internal/cli/deployments/diagnostics.go:67]: https://github.com/mongodb/mongodb-atlas-cli/blob/010d10c5fdcc06ff12f08e3b5d15189a2265e9ff/test/e2e/atlas/deployments_local_auth_test.go#L50-L62

File: internal/container/docker.go (hunk 1 above)
Comment (fmenezes) [reply internal/container/docker.go:404]: added

**Review by blva — APPROVED** [review]

**Pull request merged**